### PR TITLE
FB 5403: Deprecate RedirectAction for SimpleRedirectAction

### PR DESCRIPTION
**Pull request by RosalineP**

### CODE REVIEW

File: api/src/org/labkey/api/action/SpringActionController.java [hunk 1]
```diff
@@ -1195,7 +1195,7 @@ public static void executingMutatingSql(String sql)
         {
             readonly = true;
         }
-        else if (SimpleRedirectAction.class.isAssignableFrom(c) || RedirectAction.class.isAssignableFrom(c) || SimpleViewAction.class.isAssignableFrom(c))
+        else if (SimpleRedirectAction.class.isAssignableFrom(c) || SimpleViewAction.class.isAssignableFrom(c))
```
Comment (labkey-adam) [inline api/src/org/labkey/api/action/SpringActionController.java]: Should leave this RedirectAction check in place until we actually delete it.

File: api/src/org/labkey/api/query/LookupForeignKey.java [hunk 2]
```diff
@@ -168,7 +168,7 @@ protected StringExpression getURL(ColumnInfo parent, boolean useDetailsURL)
     {
         if (null != _baseURL)
         {
-            // CONSIDER: set ContainerContext in AbstractForeignKey.getURL() so all subclasses can benefit
+            // CONSIDER: set ContainerContext in AbstractForeignKey.getRedirectURL() so all subclasses can benefit
```
Comment (labkey-adam) [inline api/src/org/labkey/api/query/LookupForeignKey.java]: Revert this change... looks unrelated to redirect actions (and will confuse future readers)

File: api/src/org/labkey/api/query/LookupForeignKey.java [hunk 3]
```diff
@@ -207,7 +207,7 @@ public static StringExpression getDetailsURL(ColumnInfo parent, TableInfo lookup
             else
                 rewrite = f.remapFieldKeys(parent.getFieldKey(), null);
 
-            // CONSIDER: set ContainerContext in AbstractForeignKey.getURL() so all subclasses can benefit
+            // CONSIDER: set ContainerContext in AbstractForeignKey.getRedirectURL() so all subclasses can benefit
```
Comment (labkey-adam) [inline api/src/org/labkey/api/query/LookupForeignKey.java]: Revert this change... looks unrelated to redirect actions (and will confuse future readers)

File: core/src/org/labkey/core/login/LoginController.java [hunk 4]
```diff
@@ -2201,10 +2201,9 @@ public NavTree appendNavTrail(NavTree root)
     @RequiresNoPermission
     // This action has historically accepted GET. Technically, it is a mutating operation, but only in the case
     // where the caller has a secrete (the authentication token).
```
Comment (labkey-adam) [inline core/src/org/labkey/core/login/LoginController.java]: While you're in here, maybe you could correct my misspelling ("secrete"->"secret") :)

File: core/src/org/labkey/core/portal/ProjectController.java [hunk 5]
```diff
@@ -1543,7 +1542,7 @@ public NavTree appendNavTrail(NavTree root)
 
 
     @RequiresNoPermission
-    public class RedirectAction extends SimpleViewAction<ReturnUrlForm>
+    public class ProjectRedirectAction extends SimpleViewAction<ReturnUrlForm>
```
Comment (labkey-adam) [inline core/src/org/labkey/core/portal/ProjectController.java]: I don't think you meant to rename this action... it's a different RedirectAction. Do you know who calls it? (A quick look make me think it's completely unused.)

Comment (labkey-adam) [reply core/src/org/labkey/core/portal/ProjectController.java]: Actually, unless you see something using it, just remove this action entirely. It's the world's dumbest action.

File: query/src/org/labkey/query/QueryDefinitionImpl.java [hunk 6]
```diff
@@ -539,7 +541,10 @@ public TableInfo getTable(@NotNull UserSchema schema, @Nullable List<QueryExcept
             TableInfo table = _cache.get(key);
             if (table == null)
             {
-                table = createTable(schema, errors, includeMetadata, null, skipSuggestedColumns, allowDuplicateColumns);
+                try (var ignored = SpringActionController.ignoreSqlUpdates())
+                {
+                    table = createTable(schema, errors, includeMetadata, null, skipSuggestedColumns, allowDuplicateColumns);
+                }
```
Comment (labkey-adam) [inline query/src/org/labkey/query/QueryDefinitionImpl.java]: Does this change belong in the CSRF branch instead?

File: query/src/org/labkey/query/controllers/QueryController.java [hunk 7]
```diff
@@ -5881,7 +5880,6 @@ public URLHelper getURL(QueryExportAuditForm form, Errors errors)
             return url;
         }
 
-        @Override
         public void validateCommand(QueryExportAuditForm form, Errors errors)
```
Comment (labkey-adam) [inline query/src/org/labkey/query/controllers/QueryController.java]: The fact that you had to remove this @Override means this code will never be called any more. IntelliJ should also show you that no one is calling this code. Important not to orphan code like this. Simple fix is to move this code into the first line of getRedirectUrl().

File: api/src/org/labkey/api/action/SpringActionController.java (hunk 1 above)
Comment (RosalineP) [reply api/src/org/labkey/api/action/SpringActionController.java]: Done!

File: api/src/org/labkey/api/query/LookupForeignKey.java (hunk 2 above)
Comment (RosalineP) [reply api/src/org/labkey/api/query/LookupForeignKey.java]: Done!

File: core/src/org/labkey/core/login/LoginController.java (hunk 4 above)
Comment (RosalineP) [reply core/src/org/labkey/core/login/LoginController.java]: No more caller secretions!

File: query/src/org/labkey/query/controllers/QueryController.java (hunk 7 above)
Comment (RosalineP) [reply query/src/org/labkey/query/controllers/QueryController.java]: Oops! Got it, thank you. :)

File: query/src/org/labkey/query/QueryDefinitionImpl.java (hunk 6 above)
Comment (RosalineP) [reply query/src/org/labkey/query/QueryDefinitionImpl.java]: So it does, thanks for the catch!

**Pull request merged**